src/Microsoft.ML.TimeSeries/STL/BackendStructureBase.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System.Collections.Generic;
+
+namespace Microsoft.ML.TimeSeries
+{
+    public enum TimeSeriesInfoKind
+    {
+        /// <summary>
+        /// (single-time series) the outliers of a single time series
+        /// </summary>
+        Outlier,
+
+        /// <summary>
+        /// (single-time series) the seasonal signal of a single time series
+        /// </summary>
+        Seasonal,
+
+        /// <summary>
+        /// (single-time series) the trend curve of a single time series
+        /// </summary>
+        Trend,
+
+        /// <summary>
+        /// (two-time series) the lead/lag correlation between two time series
+        /// </summary>
+        CrossCorrelation,
+
+        /// <summary>
+        /// (two-time series) the correlation of outliers from two time series
+        /// </summary>
+        OutlierCorrelation,
+
+        /// <summary>
+        /// (two-time series) the correlation of trends from two time series
+        /// </summary>
+        TrendCorrelation,
+    }
+
+    public abstract class TimeSeriesInfoBase
+    {
+        /// <summary>
```

Comment:
So at first look I'm confused about when do you use this class and its derived classes (abstract SingleSeriesInfo, SeasonalInfo and TrendInfo)?. The concrete classes SeasonalInfo and TrendInfo do have a constructor, but when looking in VisualStudio it says that it has 0 references... so it seems you're actually never instantiating any the concrete children of TimeSeriesInfoBase, and so it would seem to me these classes are actually never used? Perhaps I'm missing something 😕 

Also the static Compare method bellow has 0 references.

Reply:
If there's a reason to keep these classes, please consider making them `internal` instead of `public` for the reasons I've explained here:
[https://github.com/dotnet/machinelearning/pull/5202/files?file-filters%5B%5D=.cs&file-filters%5B%5D=.csv#r440495855](https://github.com/dotnet/machinelearning/pull/5202/files?file-filters%5B%5D=.cs&file-filters%5B%5D=.csv#r440495855)

Thanks.

---
In reply to: [440494174](https://github.com/dotnet/machinelearning/pull/5202#discussion_r440494174) [](ancestors = 440494174)

Reply:
removed these unused classes.

---
In reply to: [440496618](https://github.com/dotnet/machinelearning/pull/5202#discussion_r440496618) [](ancestors = 440496618,440494174)